pylint/checkers/format.py

```diff
@@ -382,19 +383,15 @@ def _check_keyword_parentheses(
                     # The empty tuple () is always accepted.
                     if i == start + 2:
                         return
-                    if keyword_token == "not":
-                        if not found_and_or:
-                            self.add_message(
-                                "superfluous-parens", line=line_num, args=keyword_token
-                            )
-                    elif keyword_token in {"return", "yield"}:
-                        self.add_message(
-                            "superfluous-parens", line=line_num, args=keyword_token
-                        )
-                    elif not found_and_or and keyword_token != "in":
-                        self.add_message(
-                            "superfluous-parens", line=line_num, args=keyword_token
-                        )
+                    if found_and_or:
+                        return
+                    if keyword_token == "in":
+                        # This special case was added in https://github.com/PyCQA/pylint/pull/4948
+                        # but it could be removed in the future. Avoid churn for now.
+                        return
```

Comment:
@DanielNoord this is here to preserve the special case added in #4948, but when I looked into it, I had trouble figuring out why it was deemed a bugfix. (Looks like unnecessary parentheses to me!) Do you recall the thought behind it?

Reply:
Does this still pass:
`D = [x for x in ((3, 4) if 1 > 0 else ((5, 6)))]`

See https://github.com/PyCQA/pylint/issues/4907#issuecomment-907640922

I think sometimes people want to create inner tuples and this was added so we don't flag those.

Reply:
Yes, it still passes now, but I'm essentially suggesting we later revert #4948.

> I think sometimes people want to create inner tuples and this was added so we don't flag those.

I'm not sure what _inner tuple_ means here. `((5,6),)` is a tuple of a tuple. `((5,6))` is superfluous parentheses, so it could just be disabled if people really prefer the extra parentheses. But I think maybe there's a nuance here I'm not getting?

Reply:
> Does this still pass:
D = [x for x in ((3, 4) if 1 > 0 else ((5, 6)))]

(This test is still in the suite)

Reply:
Hmm perhaps I was wrong to re-allow those.

I thought there might be use cases where you would want to create `Tuple[Tuple[Tuple[int, ...]]]` instead of `Tuple[int, ...]`?

Reply:
Sure, but that's `(((5,),),)` rather than `(((5,)))`.

Parentheses increase readability, so I understand why people use them, belt and suspenders and all, but I'm just saying those folks might not want the check enabled if they're in the practice of adding them.

Reply:
Hm, I'm fine with reverting then. But we should check #4907 for any legitimate use cases.

Reply:
I'll capture this discussion on #5361